packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart

```diff
@@ -117,29 +118,85 @@ class _AllocationTracingTableState extends State<AllocationTracingTable> {
-          sortDirection: sortDirection,
-          selectionNotifier: widget.controller.selectedTracedClass,
-          onSortChanged: (column, direction, {secondarySortColumn}) {
-            // Keep track of sorting state so it doesn't get reset when
-            // `controller.refreshing` changes.
-            setState(() {
-              sortDirection = direction;
-              secondarySortColumn = secondarySortColumn;
-            });
+    return Column(
+      children: [
+        Padding(
+          padding: const EdgeInsets.only(bottom: densePadding),
+          child: ClassFilterTextField(
+            controller: widget.controller,
+          ),
+        ),
+        DualValueListenableBuilder<bool, List<TracedClass>>(
+          firstListenable: widget.controller.refreshing,
+          secondListenable: widget.controller.classList,
+          builder: (context, _, classList, __) {
+            return Expanded(
+              child: OutlineDecoration(
+                child: FlatTable<TracedClass>(
+                  columns: columns,
+                  data: classList,
+                  keyFactory: (e) => Key(e.cls.id!),
+                  onItemSelected: widget.controller.selectTracedClass,
+                  sortColumn: _checkboxColumn,
+                  secondarySortColumn: secondarySortColumn,
+                  sortDirection: sortDirection,
+                  selectionNotifier: widget.controller.selectedTracedClass,
+                  onSortChanged: (column, direction, {secondarySortColumn}) {
+                    // Keep track of sorting state so it doesn't get reset when
+                    // `controller.refreshing` changes.
+                    setState(() {
+                      sortDirection = direction;
+                      secondarySortColumn = secondarySortColumn;
+                    });
+                  },
+                ),
+              ),
+            );
           },
-        );
-      },
+        ),
+      ],
+    );
+  }
+}
+
+class ClassFilterTextField extends StatelessWidget {
+  ClassFilterTextField({
+    Key? key,
+    required this.controller,
+  }) : super(key: key);
+
+  final AllocationProfileTracingViewController controller;
+  final _textEditingController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return TextField(
+      controller: _textEditingController,
+      decoration: InputDecoration(
+        constraints: const BoxConstraints(
+          // Forces the row and clear icon to be dense. Default is 48, as per
+          // Flutter documentation. For more details, see:
+          // https://api.flutter.dev/flutter/material/InputDecoration/prefixIconConstraints.html
+          minHeight: 32,
+          maxHeight: 32,
+        ),
+        contentPadding: const EdgeInsets.all(denseSpacing),
+        border: const OutlineInputBorder(),
+        labelText: 'Class Filter',
+        hintText: 'Filter by class name',
+        suffixIcon: IconButton(
```

Comment:
can you use `suffix` and use the `clearInputButton` helper here instead?

Reply:
To be honest, I don't think it looks quite right as it's not correctly aligned and isn't visible unless the filter text field is focused. I think we should consider switching to this style, using `suffixIcon: clearInputButton(() {...})` where `clearInputButton` returns this widget.